paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/PaimonCommand.scala

```diff
@@ -112,6 +114,14 @@ trait PaimonCommand extends WithFileStoreTable with ExpressionHelper {
       sparkSession: SparkSession): Array[String] = {
     import sparkSession.implicits._
 
+    // only raw convertible can generate input_file_name()
+    for (split <- candidateDataSplits) {
+      if (!split.rawConvertible()) {
+        throw new IllegalArgumentException(
+          "Only compacted table can generate touched files, please use 'COMPACT' procedure.");
```

Comment:
`please use 'COMPACT' procedure` -> `please use 'COMPACT' procedure before DELETE`

Reply:
this method is shared by DELETE and UPDATE.

Reply:
add `first`